### PR TITLE
Bug 1197383 - Refactor FTU L10n API for v3

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1197383

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=28ccdc8bddbae74534b53a5e4baa6418d62fcbe3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b6b95c560abd75cf5fb20b34af0778da483700a7

File: apps/ftu/test/unit/language_test.js [hunk 1]
```diff
@@ -101,9 +101,12 @@ suite('languages >', function() {
       assert.isTrue(KeyboardHelper.changeDefaultLayouts.called);
     });
 
-    test('localize changed', function() {
+    test('localize changed', function(done) {
```
Comment (sfoster) [inline apps/ftu/test/unit/language_test.js]: Nit: this way the test will time out of the assertion fails (throws) (yes I know there are other tests in this file with the same problem). We want to move towards something like:

      return Promise.resolve().then(() => {
        assert.equal(MockNavigatorSettings.mSettings['locale.hour12'], false);
      }).then(done, done);

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5879ced22eb8b45c60bbe0b2ba115be935528d65

File: apps/ftu/test/unit/language_test.js (hunk 1 above)
Comment (zbraniecki) [inline apps/ftu/test/unit/language_test.js]: thanks!

**Pull request merged**